app/assets/javascripts/file_manager.js

```diff
@@ -0,0 +1,137 @@
+    $.ajax({
+        url: "/file_manager/" + rel_path,
+        type: "PUT",
+        data: { new_name: new_name, relative_path: rel_path},
+        success: function(data) {
+            console.log(`Renamed: ${rel_path}`)
+            location.reload();
+        }
+    });
+}
+
+function deleteSelected(path) {
+    $.ajax({
+        url: path,
+        type: "DELETE",
+        success: function () {
+            console.log(`Deleted: ${path}`);
+            location.reload();
+        },
+        error: function (xhr, status, error) {
+            console.error(`Failed to delete ${path}: ${error}`);
+        }
+    });
+}
+
+function downloadSelected(path) {
+    $.ajax({
+        url: '/file_manager/download_tar/',
+        type: "POST",
+        data: {
+            path: path
+        },
+        success: function (data) {
+            let blob = new Blob([data], { type: 'application/x-tar' });
+            let url = URL.createObjectURL(blob);
+            let a = document.createElement('a');
+            a.href = url;
+            let parts = path.split("/")
+            a.download = parts[parts.length - 1]
+            a.style.display = 'none';
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
+            URL.revokeObjectURL(url);
+            console.log(`Downloaded: ${data.filename}`);
+        },
+        error: function (xhr, status, error) {
+            console.error(`Failed to download ${path}: ${error}`);
+        }
+    });
+}
+
+function uploadFile(file, path, name) {
+    let formData;
+    let pathSegments = path.split('/');
+    pathSegments.pop();
+    let modifiedPath = pathSegments.join('/');
+    formData = new FormData();
+    formData.append('file', file);
+    formData.append('name', name);
+    $.ajax({
+        url: modifiedPath,
+        type: "POST",
+        data: formData,
+        contentType: false,
+        processData: false,
+        success: function () {
+            console.log("Uploaded files successfully");
+            location.reload();
+        },
+        error: function (xhr, status, error) {
+            console.error(`Failed to upload files: ${error}`);
+        }
+    });
+}
+
+function uploadAllFiles(path) {
+    let inputElement = document.getElementById('fileInput');
+    let files = inputElement.files;
+    if (files.length > 0) {
+        for (let i = 0; i < files.length; i++) {
+            let file = files[i];
+            uploadFile(file, path, "");
+        }
+    } else {
+        console.log('No files selected.');
+    }
+}
+
+function getSelectedItems() {
+    let selectedItems = $("input[type='checkbox']:checked");
+    let paths = [];
+    selectedItems.each(function(index, element) {
+        paths.push(jQuery(element).prop('value'));
+    });
+    return paths
+}
+
+function createFolder(path) {
+    let name = prompt("Name of folder: ");
+    if (name !== "") {
```

Comment:
Should compare against `null`, in case user clicks "cancel"

Reply:
<img width="301" alt="Screenshot 2024-03-05 at 22 33 37" src="https://github.com/autolab/Autolab/assets/9074856/c3c3fd5d-d871-4c7a-b554-1c17d4c477d3">

Reply:
fixed.